crates/subspace-service/src/piece_cache.rs

```diff
@@ -55,28 +107,37 @@ where
 
     /// Add pieces to cache (likely as the result of archiving)
     pub fn add_pieces(
-        &self,
+        &mut self,
         first_piece_index: PieceIndex,
         pieces: &FlatPieces,
     ) -> Result<(), Box<dyn Error>> {
         if self.max_pieces_in_cache == 0 {
             return Ok(());
         }
 
-        let insert_keys = (first_piece_index..)
+        let insert_indexes = (first_piece_index..)
             .take(pieces.count())
-            .map(Self::key)
             .collect::<Vec<_>>();
```

Comment:
It makes sense to collect this to `BTreeSet` here, which will allow more efficient insertion into another `BTreeSet` below: https://doc.rust-lang.org/stable/std/collections/struct.BTreeSet.html#method.append